lib/chef/provider/package/yum.rb

```diff
@@ -1054,15 +1054,17 @@ def load_current_resource
           # 3) or a dependency, eg: "foo >= 1.1"
 
           # Check if we have name or name+arch which has a priority over a dependency
-          package_name_array.each do |n|
+          package_name_array.each_with_index do |n,index|
             unless @yum.package_available?(n)
               # If they aren't in the installed packages they could be a dependency
-              dep = parse_dependency(n)
+              dep = parse_dependency(n,new_version_array[index])
```

Comment:
style nit - both ruby style and this code base, use `, ` between arguments to functions, but in the call and in the prototype, but in your diff, you consistently don't use the space (`foo(bar, baz)` vs `foo(bar,baz)`)